File: app/src/main/java/org/torproject/android/ui/dialog/AboutDialogFragment.java
package org.torproject.android.ui.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import androidx.fragment.app.DialogFragment;

import org.torproject.android.R;
import org.torproject.android.BuildConfig;
import org.torproject.android.core.DiskUtils;
import org.torproject.android.service.OrbotService;

import java.io.IOException;

import IPtProxy.IPtProxy;

public class AboutDialogFragment extends DialogFragment {

    public static final String TAG = AboutDialogFragment.class.getSimpleName();
    public static final String VERSION = BuildConfig.VERSION_NAME;
    private static final String BUNDLE_KEY_TV_ABOUT_TEXT = "about_tv_txt";
    private TextView tvAbout;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        View view = getActivity().getLayoutInflater().inflate(R.layout.layout_about, null);

        TextView versionName = view.findViewById(R.id.versionName);
        versionName.setText(VERSION);

        tvAbout = view.findViewById(R.id.aboutother);

        TextView tvObfs4 = view.findViewById(R.id.tvObfs4);
        tvObfs4.setText(getString(R.string.obfs4_url, IPtProxy.lyrebirdVersion()));

        TextView tvTor = view.findViewById(R.id.tvTor);
        tvTor.setText(getString(R.string.tor_url, OrbotService.BINARY_TOR_VERSION));

        TextView tvSnowflake = view.findViewById(R.id.tvSnowflake);
        tvSnowflake.setText(getString(R.string.snowflake_url, IPtProxy.snowflakeVersion()));

        boolean buildAboutText = true;

        if (savedInstanceState != null) {
            String tvAboutText = savedInstanceState.getString(BUNDLE_KEY_TV_ABOUT_TEXT);
            if (tvAboutText != null) {
                buildAboutText = false;
                tvAbout.setText(tvAboutText);
            }
        }

        if (buildAboutText) {
            try {
                String aboutText = DiskUtils.readFileFromAssets("LICENSE", getContext());
                aboutText = aboutText.replace("\n", "<br/>");
                tvAbout.setText(Html.fromHtml(aboutText));
            } catch (IOException e) {
            }
        }
        return new AlertDialog.Builder(getContext())
                .setTitle(getString(R.string.button_about))
                .setView(view)
                .create();
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString(BUNDLE_KEY_TV_ABOUT_TEXT, tvAbout.getText().toString());
    }
}

File: app/src/main/java/org/torproject/android/ui/more/AboutDialogFragment.kt
package org.torproject.android.ui.more

import IPtProxy.IPtProxy
import android.app.Dialog
import android.graphics.Typeface
import android.os.Bundle
import android.text.SpannableStringBuilder
import android.text.Spanned
import android.text.style.StyleSpan
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import org.torproject.android.R
import org.torproject.android.databinding.LayoutAboutBinding
import org.torproject.android.util.DiskUtils
import org.torproject.android.util.createWithCurves
import org.torproject.jni.BuildConfig.VERSION_NAME
import org.torproject.jni.TorService
import java.io.IOException

class AboutDialogFragment : DialogFragment() {

    companion object {
        const val TAG = "AboutDialogFragment"
        const val VERSION: String = VERSION_NAME
        private const val BUNDLE_KEY_TV_ABOUT_TEXT = "about_tv_txt"
    }

    private lateinit var binding: LayoutAboutBinding

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = LayoutAboutBinding.inflate(layoutInflater)
        val versionName = view?.findViewById<TextView>(R.id.versionName)
        versionName?.text = VERSION

        binding.tvTor.text = getString(R.string.tor_url, TorService.VERSION_NAME)
        binding.tvObfs4.text =
            getString(R.string.obfs4_url, IPtProxy.lyrebirdVersion().substringAfter('-'))
        binding.tvSnowflake.text = getString(R.string.snowflake_url, IPtProxy.snowflakeVersion())

        var buildAboutText = true

        savedInstanceState?.getString(BUNDLE_KEY_TV_ABOUT_TEXT)?.let {
            buildAboutText = false
            binding.aboutother.text = it
        }

        if (buildAboutText) {
            try {
                val equalsBlockRegex = Regex("={3,}")
                var aboutText = DiskUtils.readFileFromAssets("LICENSE", requireContext())
                aboutText = aboutText.replace(equalsBlockRegex, "")

                val spannableAboutText = SpannableStringBuilder(aboutText)
                spannableAboutText.setSpan(
                    StyleSpan(Typeface.BOLD),
                    0,
                    aboutText.indexOf("\n"),
                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
                )

                binding.aboutother.text = spannableAboutText
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }

        return AlertDialog.Builder(requireContext(), R.style.OrbotDialogTheme)
            .setTitle(getString(R.string.menu_about))
            .setView(binding.root)
            .createWithCurves()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString(BUNDLE_KEY_TV_ABOUT_TEXT, binding.aboutother.text.toString())
    }
}